library/pkwrite.c

```diff
@@ -253,6 +264,8 @@ int mbedtls_pk_write_pubkey_der(const mbedtls_pk_context *key, unsigned char *bu
         mbedtls_svc_key_id_t key_id;
         psa_ecc_family_t curve;
         size_t bits;
+        size_t oid2_len = 0;
+        const char *oid2;
```

Comment:
Note commenting on this specific line, but on the whole ` if (pk_type == MBEDTLS_PK_OPAQUE) {` block: this needs adapting to handle curdle curves as well.

I think instead of `mbedtls_psa_get_ecc_oid_from_id()` we should use `mbedtls_ecc_group_of_psa()` and then we can use `mbedtls_oid_get_oid_by_ec_grp_algid()` as above, and even `pk_write_ec_param()` if we just change its signature to take the group id as the last argument instead of the whole `ecp_keypair` which it doesn't need.

Note that currently we don't have unit-testing for this in `test_suite_pkwrite` - it's only testing indirectly in `test_suite_x509write` so far. This is embarrassing, it's the 2nd time we run into a pre-existing test gap while reviewing your PR. I'll create a separate issue for that.

Reply:
See https://github.com/Mbed-TLS/mbedtls/issues/7480